### DISCUSSION TITLE
some unapprehensive  concept

### DISCUSSION THREAD

**Discussion by shiyuleixia** [ProjectOpenSea/seaport#549] [Q&A] [answered]
1. what's the meaning for criteria-based or not criteria-based
2. what's the difference between zone and conduit
3. I read through your reference doc many times ，found some about "item" not qualified by "offer" or consideration,so I really don't understand where the item is include in,such as "All other items have the same Ether (or other native tokens) or ERC20 item type and token."
4. so,the repo is just for fulfill an order ,but not including creating an order ?

**Comment by Osub** [comment]
me too

**Comment by caiiiyua** [comment]
Seems like `conduit` is a substitute of proxy account of legacy opeansea contract, it will get result of `IsApproval` and also the target of `SetApprovalForAll`. 
I also confused about `Zone` atm.

Btw, the default `conduitkey` in seaport-js sdk is `0`, in this case, the corresponding `conduit` address is `0x00000000006c3852cbEf3e08E8dF289169EdE581` which is seaport contract address, I believe this is wrong. The correct `conduit` address should be `0x1E0049783F008A0085193E00003D00cd54003c71`

**Comment by ATV3320** [comment, marked as answer]
Even though I'm new to this but I've been reading the docs and skimming through the code and test cases, so I'll try to answer them as well:

1. Criteria is for the nonfungible items only (ERC721, ERC1155), so let's say you want purpleSteel NFT(example) number 1 to 20 in exchange for whatever you're offering, but not others, but you have only passed the contract address of purpleSteel NFT, so you'd receive offers from NFTs of tokenID other than 1 to 20 as well, to prevent this, we have token identifiers, so you'd create a Merkle root composed of all the valid set of tokens (1 to 20) so that when a fulfiller(buyer) comes to fulfill, they'd just give proof that they have an NFT that has a token ID that is in the Merkle root, and then the transaction would proceed. If you just want any NFT of purpleSteel contract, you'd not provide any identifier.
2. Why is the zone there, I have less idea of that, but what can it do, we can just find it in the overview of seaport docs, so the zone is an optional secondary account for an order which has two privileges, it may cancel the order, and in case of restricted offers, it can execute the order just like the offerer. Conduit on the other hand is the contract that we as buyers/sellers are giving our tokens' approvals to, so that conduit can perform the transfer during fulfillment.
3. I don't see this..
4. Creating the orders would take place on front end, and only after fulfilling would it go on-chain, this is why we have only functions to fulfill orders.

**Reply by nfts4f4fff** [reply]
Hey there,
You can't create a merkle proof with just the root.the token ids are not given by opensea. Any idea why that is and how I can make a proper merkle proof ?

**Reply by ryanio** [reply]
@nfts4f4fff https://docs.opensea.io/reference/fulfill-an-offer

**Reply by nfts4f4fff** [reply]
It doesn't give me the criteria proof lol. Useless still

**Comment by 0xAskar** [comment]
1. unsure 
2. zone is some higher level second address that has special permissions. I don't know much details on that
3. I don't quite understand your question
4. There is no createOrder function in the smartContract I believe; however, they give us the structs so you can create it yourself. If you're using JS, you can use [seaport-js](https://github.com/ProjectOpenSea/seaport-js/blob/main/src/seaport.ts) use the function `createOrder`. But yes, you can construct your own order by passing in the correct parameters, then you can use that to `validate`, `fulfillOrder`, etc...
